internal/provider/resource_definition_criteria_resource.go

```diff
@@ -0,0 +1,273 @@
+	}
+
+	httpResp, err := r.client().GetOrgsOrgIdResourcesDefsDefIdWithResponse(ctx, r.orgId(), data.ResourceDefinitionID.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError(HUM_CLIENT_ERR, fmt.Sprintf("Unable to read resource definition, got error: %s", err))
+		return
+	}
+
+	if httpResp.StatusCode() == 404 {
+		resp.Diagnostics.AddWarning("Resource definition not found", fmt.Sprintf("The resource definition (%s) was deleted outside Terraform", data.ResourceDefinitionID.ValueString()))
+		resp.State.RemoveResource(ctx)
+		return
+	}
+
+	if httpResp.StatusCode() != 200 {
+		resp.Diagnostics.AddError(HUM_API_ERR, fmt.Sprintf("Unable to read resource definition, unexpected status code: %d, body: %s", httpResp.StatusCode(), httpResp.Body))
+		return
+	}
+
+	if httpResp.JSON200 == nil {
+		resp.Diagnostics.AddWarning("Resource definition criteria not found", fmt.Sprintf("The resource definition criteria (%s) was deleted outside Terraform", data.ID.ValueString()))
+		resp.State.RemoveResource(ctx)
+		return
+	}
+
+	res := *httpResp.JSON200
+
+	if res.Criteria == nil {
+		resp.Diagnostics.AddWarning("Resource definition criteria not found", fmt.Sprintf("The resource definition criteria (%s) was deleted outside Terraform", data.ID.ValueString()))
+		resp.State.RemoveResource(ctx)
+		return
+	}
+
+	criteria := *res.Criteria
+
+	found := false
+	for _, c := range criteria {
+		if c.Id == data.ID.ValueString() {
+			found = true
+			parseResourceDefinitionCriteriaResponse(&c, data)
+		}
+	}
+
+	if !found {
+		resp.Diagnostics.AddWarning("Resource definition criteria not found", fmt.Sprintf("The resource definition criteria (%s) was deleted outside Terraform", data.ID.ValueString()))
+		resp.State.RemoveResource(ctx)
+		return
+	}
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Save updated data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *ResourceDefinitionCriteriaResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	resp.Diagnostics.AddError("UNSUPPORTED_OPERATION", "Updating an ResourceDefinitionCriteria is currently not supported")
+}
+
+func (r *ResourceDefinitionCriteriaResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var data *ResourceDefinitionCriteriaResourceModel
+
+	// Read Terraform prior state data into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	force := false
+	httpResp, err := r.client().DeleteOrgsOrgIdResourcesDefsDefIdCriteriaCriteriaIdWithResponse(ctx, r.orgId(), data.ResourceDefinitionID.ValueString(), data.ID.ValueString(), &client.DeleteOrgsOrgIdResourcesDefsDefIdCriteriaCriteriaIdParams{
+		Force: &force,
+	})
+	if err != nil {
+		resp.Diagnostics.AddError(HUM_CLIENT_ERR, fmt.Sprintf("Unable to delete definition criteria, got error: %s", err))
+		return
+	}
+
+	if httpResp.StatusCode() != 204 {
```

Comment:
This one, if possible, for 409 errors in general, add a timeout.

it is possible resources are being de provisioned while trying deleting this, and will work in a few min, so a retry is ok. Say default timeout is 3 min.

 If this add a lot of complexity now, could be done later.